src/Microsoft.VisualStudio.Services.Agent/DeploymentGroupServer.cs

```diff
@@ -15,7 +15,23 @@ public interface IDeploymentGroupServer : IAgentService
         Task<List<DeploymentGroup>> GetDeploymentGroupsAsync(string projectName, string machineGroupName);
 
         // Update Machine Group ( Used for adding tags)
-        Task<List<DeploymentMachine>> UpdateDeploymentMachinesAsync(string projectName, int machineGroupId, List<DeploymentMachine> deploymentMachines);
+        Task<DeploymentMachine> UpdateDeploymentMachineAsync(Guid projectId, int deploymentGroupId, int machineId, DeploymentMachine deploymentMachine);
```

Comment:
GetDeploymentGroup will be called for both config and un-config. 
1) With config now we will start working with project id instead of project name
2) With un-config for old configured agents will have projectName saved with them. We need to work with projectName in that case
This will be valid for DeleteMachine as well which is getting called for unconfig